test/extended/setup_compat.sh

```diff
@@ -0,0 +1,354 @@
+      openshift/origin:${API_SERVER_VERSION} start master api \
+      --config=${MASTER_CONFIG_DIR}/master-config.yaml &>"${LOG_DIR}/apiserver.log" &
+  else
+    ${sudo} env "PATH=${PATH}" OPENSHIFT_PROFILE=web OPENSHIFT_ON_PANIC=crash openshift start master api \
+      --config=${MASTER_CONFIG_DIR}/master-config.yaml \
+    &>"${LOG_DIR}/apiserver.log" &
+  fi
+  export API_SERVER_PID=$!
+
+  echo "[INFO] OpenShift API server start at: "
+  date
+
+  wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz" "apiserver: " 0.25 80
+
+  echo "[INFO] OpenShift server health checks done at: "
+  date
+}
+
+# start_api_server starts the OpenShift API server
+function start_controllers() {
+  local sudo="${USE_SUDO:+sudo}"
+
+  echo "[INFO] controller logs will be at:    ${LOG_DIR}/controller.log"
+  echo "[INFO] Config dir is:                 ${SERVER_CONFIG_DIR}"
+  echo "[INFO] MasterIP is:                   ${MASTER_ADDR}"
+
+  mkdir -p ${LOG_DIR}
+
+  echo "[INFO] Starting OpenShift controllers"
+  if [[ -n "${CONTROLLER_VERSION:-}" ]]; then
+    ${sudo} docker run -e OPENSHIFT_ON_PANIC=crash \
+      --privileged --net=host -v ${MASTER_CONFIG_DIR}:${MASTER_CONFIG_DIR} \
+      openshift/origin:${CONTROLLER_VERSION} start master controllers \
+      --config=${MASTER_CONFIG_DIR}/master-config.yaml &>"${LOG_DIR}/controller.log" &
+  else
+    ${sudo} env "PATH=${PATH}" OPENSHIFT_ON_PANIC=crash openshift start master controllers \
+      --config=${MASTER_CONFIG_DIR}/master-config.yaml \
+    &>"${LOG_DIR}/controller.log" &
+  fi
+  export CONTROLLER_PID=$!
+
+  echo "[INFO] OpenShift controllers start at: "
+  date
+}
+
+# start_node starts the OpenShift node
+function start_node() {
+  local sudo="${USE_SUDO:+sudo}"
+
+  local use_latest_images
+  if [[ -n "${USE_LATEST_IMAGES:-}" ]]; then
+          use_latest_images="true"
+  else
+          use_latest_images="false"
+  fi
+
+  echo "[INFO] node logs will be at:      ${LOG_DIR}/node.log"
+  echo "[INFO] Volumes dir is:            ${VOLUME_DIR}"
+  echo "[INFO] Config dir is:             ${SERVER_CONFIG_DIR}"
+  echo "[INFO] Using images:              ${USE_IMAGES}"
+
+  mkdir -p ${LOG_DIR}
+
+  echo "[INFO] Starting OpenShift node"
+  ${sudo} env "PATH=${PATH}" OPENSHIFT_ON_PANIC=crash openshift start node \
+    --config=${NODE_CONFIG_DIR}/node-config.yaml \
+    --loglevel=4 --logspec='*importer=5' \
+    --latest-images="${use_latest_images}" \
+  &>"${LOG_DIR}/node.log" &
+  export NODE_PID=$!
+
+  echo "[INFO] OpenShift server start at: "
+  date
+
+  wait_for_url "${KUBELET_SCHEME}://${KUBELET_HOST}:${KUBELET_PORT}/healthz" "[INFO] kubelet: " 0.5 120
+
+  echo "[INFO] OpenShift node health checks done at: "
+  date
+}
+
+# configure_versioned_os_server will configure the OpenShift server using either
+# API_SERVER_VERSION or CONTROLLER_VERSION (whichever is set)
+function configure_versioned_os_server() {
```

Comment:
isn't our normal setup just a special case of this? I think we'd be better off making the code in `hack/util.sh` more general so it supports both